tests/test_recreation.py

```diff
@@ -844,8 +842,8 @@ def test_square_grid_regression(self):
         expected_grid_vector_path = os.path.join(
             REGRESSION_DATA, 'square_grid_vector_path.shp')
 
-        pygeoprocessing.testing.assert_vectors_equal(
-            out_grid_vector_path, expected_grid_vector_path, 1E-6)
+        _assert_vector_attributes_eq(

```

Comment:
I see this function already existed and was used elsewhere in this file, basically as a means to regression test final model results. But when this function replaces `pygeoprocessing.testing.assert_vectors_equal` then we're no longer asserting equal geometries. Honestly I think that's fine for most of these tests, but this one in particular is testing the `_grid_vector` function, so it is the geometry the needs to be tested.

If we do add a `utils._assert_vectors_equal`, then this could be another use-case.